rllib/examples/learners/mog_learner.py

```diff
@@ -0,0 +1,134 @@
+the same hyperparameters. Some hiccups have been the alphas going too small which causes nans in the loss equation.
+This has been fixed by using the log_softmax(alphas) instead of using the softmax on the alphas once out of
+the module.
+
+With --num_guassians=2 and --lr=0.0001
+(trying to reach 250.0 return on CartPole in 100k env steps):
++-----------------------------+------------+-----------------+--------+
+| Trial name                  | status     | loc             |   iter |
+|                             |            |                 |        |
+|-----------------------------+------------+-----------------+--------+
+| PPO_CartPole-v1_4a3a0_00000 | TERMINATED | 127.0.0.1:54584 |     10 |
++-----------------------------+------------+-----------------+--------+
++------------------+------------------------+---------------------+
+|   total time (s) | num_env_steps_sampled_ | episode_return_mean |
+|                  |              _lifetime |                     |
+|------------------+------------------------+---------------------+
+|          42.9956 |                 40_000 |              271.53 |
++------------------+------------------------+---------------------+
+"""
+
+from pprint import pprint
+from ray.rllib.core.rl_module.rl_module import RLModuleSpec
+from ray.rllib.examples.learners.custom_ppo_config import CustomPPOConfig
+from ray.rllib.examples.rl_modules.classes.custom_mog import MOGModule
+from ray.rllib.examples.learners.classes.mog_loss import PPOTorchLearnerCustomMOGLoss
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.test_utils import (
+    add_rllib_example_script_args,
+)
+
+torch, _ = try_import_torch()
+
+
+parser = add_rllib_example_script_args(
+    default_reward=250.0,
+    default_timesteps=200_000,
+)
+parser.set_defaults(enable_new_api_stack=True)
+parser.add_argument(
+    "--num_gaussians",
+    type=int,
+    default=3,
+    help="Number of gaussians to represent the value function",
+)
+parser.add_argument(
+    "--lr",
+    type=float,
+    default=0.0003,
+    help="Learning rate",
+)
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    custom_config = {
+        "vf_share_layers": False,
+        "fcnet_hiddens": [128, 128],
+        "fcnet_activation": "LeakyReLU",
+        "num_mixture_components": args.num_gaussians,
+    }
+
+    module_to_load_spec = RLModuleSpec(
+        module_class=MOGModule,
+        model_config=custom_config,
+    )
+
+    assert (
+        args.enable_new_api_stack
+    ), "Must set --enable-new-api-stack when running this script!"
+    assert args.algo == "PPO", "Must set --algo=PPO when running this script!"
+
+    base_config = (
+        CustomPPOConfig()
+        .environment("CartPole-v1")
+        .training(
+            # set the learner class which will use the custom loss function we want
+            learner_class=PPOTorchLearnerCustomMOGLoss,
```

Comment:
Awesome! Small nit: by overriding `Algorithm.get_default_learner` and `Algorithm.get_default_rl_module_spec` you can get rid of these extra arguments. Take for example a look at DQN: https://github.com/simonsays1980/ray/blob/348589b4942768863d14968900b4d9e7d1340e9d/rllib/algorithms/dqn/dqn.py#L531